demo/full/scripts/controllers/ContentList.tsx

```diff
@@ -316,68 +347,18 @@ function ContentList({
   const [isLowLatencyChecked, setIsLowLatencyChecked] = React.useState(false);
 
   /**
-   * Load the given displayed content through the player.
+   * Load the given content through the player.
    * @param {Object|null} content
    */
-  const loadContent = React.useCallback((content: IDisplayedContent) => {
-    const {
-      url,
-      transport,
-      fallbackKeyError,
-      fallbackLicenseRequest,
-      isLowLatency,
-      drmInfos,
-    } = content;
-
-    getKeySystemsOption(drmInfos ?? [], {
-      fallbackKeyError: !!fallbackKeyError,
-      fallbackLicenseRequest: !!fallbackLicenseRequest,
-    }).then((keySystems) => {
-      loadVideo({
-        url: url ?? "",
-        transport: transport ?? "",
-        textTrackMode: "html",
-        lowLatencyMode: isLowLatency,
-        keySystems,
-      });
-    }, () => {
+  const loadContent = React.useCallback((content: IPlayableContent) => {
+    getLoadVideoOptions(content).then(
+      (loadVideoOptions) => {
+        loadVideo(loadVideoOptions);
```

Comment:
I didn't look too much into it yet but have you verified that this code is not sensible to race conditions?

For example, if `loadContent` is called multiple consecutive times, are we sure that the last call will lead to playback in the end?

Will look more into it, but that's a question I was asking myself reading this

Reply:
I tried to provoke a race condition but didn't succeed in doing so.
I still have added the code to prevent any race condition, let me know what you think of that because I'm not 100% sure that this is the correct way of doing so.